R/ConceptSetDiagnostics.R

```diff
@@ -0,0 +1,2295 @@
+                                  .data$json,
+                                  .data$cohortName),
+    any.missing = FALSE,
+    min.cols = 4,
+    add = errorMessage
+  )
+  checkmate::reportAssertions(errorMessage)
+  
+  conceptSets <- list()
+  conceptSetCounter <- 0
+  
+  for (i in (1:nrow(cohorts))) {
+    cohort <- cohorts[i,]
+    sql <-
+      extractConceptSetsSqlFromCohortSql(cohortSql = cohort$sql)
+    json <-
+      extractConceptSetsJsonFromCohortJson(cohortJson = cohort$json)
+    
+    if (nrow(sql) == 0 || nrow(json) == 0) {
+      ParallelLogger::logInfo(
+        "Cohort Definition expression does not have a concept set expression. ",
+        "Skipping Cohort: ",
+        cohort$cohortName
+      )
+    } else {
+      if (!length(sql$conceptSetId %>% unique()) == length(json$conceptSetId %>% unique())) {
+        stop(
+          "Mismatch in concept set IDs between SQL and JSON for cohort ",
+          cohort$cohortFullName
+        )
+      }
+      if (length(sql) > 0 && length(json) > 0) {
+        conceptSetCounter <- conceptSetCounter + 1
+        conceptSets[[conceptSetCounter]] <-
+          tidyr::tibble(cohortId = cohort$cohortId,
+                        dplyr::inner_join(x = sql, y = json, by = "conceptSetId"))
+      }
+    }
+  }
+  if (length(conceptSets) == 0) {
+    return(NULL)
+  }
+  conceptSets <- dplyr::bind_rows(conceptSets) %>%
+    dplyr::arrange(.data$cohortId, .data$conceptSetId)
+  
+  uniqueConceptSets <- conceptSets %>%
+    dplyr::select(.data$conceptSetExpression) %>%
+    dplyr::distinct() %>%
+    dplyr::mutate(uniqueConceptSetId = dplyr::row_number())
+  
+  conceptSets <- conceptSets %>%
+    dplyr::inner_join(uniqueConceptSets, by = "conceptSetExpression") %>%
+    dplyr::distinct() %>%
+    dplyr::relocate(.data$uniqueConceptSetId,
+                    .data$cohortId,
+                    .data$conceptSetId) %>%
+    dplyr::arrange(.data$uniqueConceptSetId,
+                   .data$cohortId,
+                   .data$conceptSetId)
+  return(conceptSets)
+}
+
+
+mergeTempTables <-
+  function(connection,
+           tableName,
+           tempTables,
+           tempEmulationSchema) {
+    valueString <-
+      paste(tempTables, collapse = "\n\n  UNION ALL\n\n  SELECT *\n  FROM ")
+    sql <-
+      sprintf("SELECT *\nINTO %s\nFROM (\n  SELECT *\n  FROM %s\n) tmp;",
+              tableName,
+              valueString)
+    sql <-
+      SqlRender::translate(sql,
+                           targetDialect = connection@dbms,
+                           tempEmulationSchema = tempEmulationSchema)
+    DatabaseConnector::executeSql(connection,
+                                  sql,
+                                  progressBar = FALSE,
+                                  reportOverallTime = FALSE)
+    
+    # Drop temp tables:
+    for (tempTable in tempTables) {
+      sql <-
+        sprintf("TRUNCATE TABLE %s;\nDROP TABLE %s;", tempTable, tempTable)
```

Comment:
use `renderTranslateExecuteSql`

Reply:
Same comment - any change would be unrelated refactor